src/main/scala/org/broadinstitute/dsde/workbench/sam/directory/LdapDirectoryDAO.scala

```diff
@@ -65,18 +65,17 @@ class LdapDirectoryDAO(
       memberDns = getAttributes(results, Attr.uniqueMember)
     } yield BasicWorkbenchGroup(WorkbenchGroupName(cn), memberDns.map(dnToSubject), WorkbenchEmail(email))
 
+  private def unmarshalGroupThrow(results: Entry): BasicWorkbenchGroup = unmarshalGroup(results).fold(s => throw new WorkbenchException(s), identity)
+
   override def loadGroups(groupNames: Set[WorkbenchGroupName]): IO[Stream[BasicWorkbenchGroup]] = {
     val filters = groupNames.grouped(batchSize).map(batch => Filter.createORFilter(batch.map(g => Filter.createEqualityFilter(Attr.cn, g.value)).asJava)).toSeq
 
-    for {
-      streamOfEither <- cs.evalOn(ecForLdapBlockingIO)(IO(ldapSearchStream(groupsOu, SearchScope.SUB, filters: _*)(unmarshalGroup)))
-      res <- streamOfEither.parSequence.fold(err => IO.raiseError(new WorkbenchException(err)), r => IO.pure(r))
-    } yield res
+    executeLdap(IO(ldapSearchStream(groupsOu, SearchScope.SUB, filters: _*)(unmarshalGroupThrow)))
   }
```

Comment:
we lose the ability to collect errors this way, but Doug suggested we don't care about collecting here